core/jobs/io/gcs_io.py

```diff
@@ -0,0 +1,198 @@
+
+        Returns:
+            PCollection. The PCollection of the number of bytes that has
+            written to GCS.
+        """
+        return (
+            file_objects
+            | 'Write files to GCS' >> beam.Map(self._write_file)
+        )
+
+    def _write_file(self, file_obj: FileObjectDict) -> int:
+        """Helper function to write file to the GCS.
+
+        Args:
+            file_obj: FileObjectDict. The dictionary having file
+                path and file data.
+
+        Returns:
+            int. Returns the number of bytes that has been written to GCS.
+        """
+        bucket = app_identity_services.get_gcs_resource_bucket_name()
+        storage_services.commit(
+            bucket, file_obj['filepath'], file_obj['data'], self.mime_type)
+        return len(file_obj['data'])
+
+
+# TODO(#15613): Here we use MyPy ignore because of the incomplete typing of
+# apache_beam library and absences of stubs in Typeshed, forces MyPy to
+# assume that PTransform class is of type Any. Thus to avoid MyPy's error (Class
+# cannot subclass 'PTransform' (has type 'Any')), we added an ignore here.
+class DeleteFile(beam.PTransform): # type: ignore[misc]
+    """Delete files from GCS."""
+
+    def expand(self, file_paths: beam.PCollection) -> beam.pvalue.PDone:
+        """Deletes the files in given PCollection.
+
+        Args:
+            file_paths: PCollection. The collection of filepaths that will
+                be deleted.
+
+        Returns:
+            PCollection. The PCollection of the file data.
+        """
+        return (
+            file_paths
+            | 'Delete the file' >> beam.Map(self._delete_file)
+        )
+
+    def _delete_file(self, file_path: str) -> None:
+        """Helper function to delete the file.
+
+        Args:
+            file_path: str. The name of the file that will be deleted.
+        """
+        bucket = app_identity_services.get_gcs_resource_bucket_name()
+        storage_services.delete(bucket, file_path)
+
+
+# TODO(#15613): Here we use MyPy ignore because of the incomplete typing of
+# apache_beam library and absences of stubs in Typeshed, forces MyPy to
+# assume that PTransform class is of type Any. Thus to avoid MyPy's error (Class
+# cannot subclass 'PTransform' (has type 'Any')), we added an ignore here.
+class GetFiles(beam.PTransform): # type: ignore[misc]
+    """Get all files with specefic prefix."""
+
+    def expand(self, prefixes: beam.PCollection) -> beam.PCollection:
+        """Returns PCollection with file names.
+
+        Args:
+            prefixes: PCollection. The collection of filepath prefixes.
+
+        Returns:
+            PCollection. The PCollection of the file names.
+        """
+        return (
+            prefixes
+            | 'Get names of the files' >> beam.Map(self._get_file_with_prefix)
+        )
+
+    def _get_file_with_prefix(
+        self, prefix: str
+    ) -> List[cloud_storage_emulator.EmulatorBlob]:
```

Comment:
Should we really return a list of blobs? Shouldn't we return just the filenames?

Reply:
Updated, will only return the names.